pandas/core/arrays/interval.py

```diff
@@ -543,33 +543,87 @@ def __eq__(self, other):
             # extract intervals if we have interval categories with matching closed
             if is_interval_dtype(other_dtype):
                 if self.closed != other.categories.closed:
-                    return np.zeros(len(self), dtype=bool)
+                    return invalid_comparison(self, other, op)
                 other = other.categories.take(other.codes)
 
         # interval-like -> need same closed and matching endpoints
         if is_interval_dtype(other_dtype):
             if self.closed != other.closed:
-                return np.zeros(len(self), dtype=bool)
-            return (self._left == other.left) & (self._right == other.right)
+                return invalid_comparison(self, other, op)
+            if isinstance(other, Interval):
+                other = type(self)._from_sequence([other])
+                if self._combined.dtype.kind in ["m", "M"]:
+                    # Need to repeat bc we do not broadcast length-1
+                    # TODO: would be helpful to have a tile method to do
+                    #  this without copies
+                    other = other.repeat(len(self))
+            else:
+                other = type(self)(other)
+
+            if op is operator.eq:
```

Comment:
were these not working before? tests on these?

Reply:
ATM IntervalArray comparison is only only defined for `__eq__`, and IntervalIndex gets its inequality comparisons from the base Index class.  We have tests for the IntervalIndex comparisons, which after this PR indirectly test the IntervalArray comparisons.  But the IntervalIndex comparison tests don't use the box_with_array pattern we use for other tests, so will need to be updated in what i expect will be a dedicated+big(ish) PR